models/bg.model.js

```diff
@@ -33,4 +33,26 @@ exports.fetchReviewById = (review_id) => {
                 return review.rows;
             }
         });
+};
+
+exports.fetchReviewComments = (review_id) => {
+    return db
+        .query(
+            `SELECT comments.* FROM comments full outer join reviews on comments.review_id = reviews.review_id WHERE reviews.review_id = $1;`,
+            [review_id]
+            //`select * from reviews;`
+        )
+        .then((review) => {
+            if (review.rows.length === 0) {
+                return Promise.reject({ status: 404, msg: "review does not exist" });
+            } else {
+                for (let i = 0; i < review.rows.length; i++) {
```

Comment:
I think this logic for determining the existence of the review is sound but definitely more complicated than it needs to be Matthew - take a look at the notes for today Advanced Error Handling for a nice solution.
But please get in touch if you have questions on it!